store/src/java/com/zimbra/cs/event/SolrEventStore.java

```diff
@@ -174,39 +180,47 @@ private String getAggregationBucketForContactFrequencyGraphTimeRange(ContactAnal
         }
     }
 
-    private LocalDateTime getStartDateForContactFrequencyGraphTimeRange(ContactAnalytics.ContactFrequencyGraphTimeRange timeRange) throws ServiceException {
+    private ZonedDateTime getStartDateForContactFrequencyGraphTimeRange(ContactAnalytics.ContactFrequencyGraphTimeRange timeRange, ZoneId userZoneId) throws ServiceException {
         switch (timeRange) {
         case CURRENT_MONTH:
-            return getStartDateForCurrentMonth();
+            return getStartDateForCurrentMonth(userZoneId);
         case LAST_SIX_MONTHS:
-            return getStartDateForLastSixMonths();
+            return getStartDateForLastSixMonths(userZoneId);
         case CURRENT_YEAR:
-            return getStartDateForCurrentYear();
+            return getStartDateForCurrentYear(userZoneId);
         default:
             throw ServiceException.INVALID_REQUEST("Time range not supported " + timeRange, null);
         }
     }
 
-    private LocalDateTime getStartDateForCurrentMonth() {
-        LocalDateTime now = LocalDateTime.now();
-        LocalDateTime firstDayOfMonth = now.with(TemporalAdjusters.firstDayOfMonth()).truncatedTo(ChronoUnit.DAYS);
-        return firstDayOfMonth;
+    private ZonedDateTime getStartDateForCurrentMonth(ZoneId userZoneId) {
+        ZonedDateTime userTZNow = ZonedDateTime.now(userZoneId);
+        ZonedDateTime userTZFirstDayOfMonth = userTZNow.with(TemporalAdjusters.firstDayOfMonth()).truncatedTo(ChronoUnit.DAYS);
+        return userTZFirstDayOfMonth;
```

Comment:
nitpick: Any reason you can't just return the `userTZNow.with(TemporalAdjusters.firstDayOfMonth()).truncatedTo(ChronoUnit.DAYS);` value directly?